components/weekly_storage/daily_storage.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_WEEKLY_STORAGE_DAILY_STORAGE_H_
+#define BRAVE_COMPONENTS_WEEKLY_STORAGE_DAILY_STORAGE_H_
+
+#include <list>
+#include <memory>
+
+#include "base/time/time.h"
+
+namespace base {
+class Clock;
+}
+
+class PrefService;
+
+// Allows to track a sum of some
+// values added from time to time via |AddDelta| over the last 24 hours.
+// Requires |pref_name| to be already registered.
+class DailyStorage {
```

Comment:
why don't we reuse weekly storage by passing an interval as a parameter? could be a template parameter basically. Note that weekly storage has unit tests. I'm not sure copypasting would be a best choice

Reply:
We possibly _could_ combine WeeklyStorage and DailyStorage, but I actually didn't copy/paste - DailyStorage is quite a bit simpler thank WeeklyStorage I think. I'm going to ask that we look in to combining them as a follow-up for consideration, as I think adding DailyStorage capability to WeeklyStorage could be seen as adding even more complexity to WeeklyStorage. Would that be acceptable to you? I've added unit tests to DailyStorage, please review.